src/main/java/org/apache/sling/feature/maven/Preprocessor.java

```diff
@@ -256,6 +305,8 @@ protected void readProjectFeatures(
                     throw new RuntimeException("Unable to read feature " + file.getAbsolutePath(), io);
                 }
 
+                checkFeatureFileValidation(file);
+
```

Comment:
If you validate the file, the validation might fail as the "id" property might not be set, this will be set by calling the next method, so validation should happen within that method

Reply:
The id is completely missing :)

Reply:
I got it - please check my latest commit, validation now happens `preprocessFeature` invocation :P